ruby/src/api.cc

```diff
@@ -157,13 +158,13 @@ namespace leatherman { namespace ruby {
         }
 
         if (_library.first_load()) {
-            // Run an empty script evaluation
-            // ruby_options is a required call as it sets up some important stuff (unfortunately)
-            char const* opts[] = {
-                "ruby",
-                "-e",
-                ""
+            // Calling ruby_sysinit to initialize windows sockets due to the change in
+            // https://github.com/ruby/ruby/commit/e33b1690d06f867e45750bd8e3e8b06d78b5bc26
+            int sysinit_opts_size = 1;
+            char const* sysinit_opts[] = {
+                "ruby"
             };
+            ruby_sysinit(&sysinit_opts_size, reinterpret_cast<char***>(&sysinit_opts));
```

Comment:
If this modification works only for ruby version >= 2.5.x, I think it should do a version check or at least documented.

Reply:
This should work regardless of ruby version. Also, this leatherman branch uses puppet-runtime 6.0.x which has ruby-2.5.3: https://github.com/puppetlabs/puppet-runtime/blob/master/configs/projects/agent-runtime-6.0.x.rb#L3

puppet-agent <= 5.5.x will not have this leatherman fix, since this issue is not present there, so we're good.